### PR TITLE
Add fadeDuration back to TextSelectionOverlay

### LINKED ISSUE #100343: Depending on packages which use `TextSelectionOverlay.fadeDuration` results in build failure
## Steps to Reproduce

If some app is depend on a library (ex. [flutter_math_fork](https://pub.dev/packages/flutter_math_fork)) that uses `TextSelectionOverlay.fadeDuration`, the app will not be build with flutter 2.12.0-4.1.pre or higher.

ref https://github.com/flutter/flutter/pull/98153

---

In the library [flutter_math_fork](https://pub.dev/packages/flutter_math_fork), `TextSelectionOverlay.fadeDuration` is used in the following line. 
https://github.com/simpleclub-extended/flutter_math_fork/blob/a6f0c766a660b38c767fb12486672165630a8a69/lib/src/widgets/selection/handle_overlay.dart#L60 

So, applications using flutter_math_fork cannot upgrade flutter until flutter_math_fork is updated.
On the other hand, if another library updates flutter to use `SelectionOverlay.fadeDuration`, the developer will need to modify that library's version. flutter_math_fork seems to be a popular library, but perhaps other libraries will have this problem as well. 
[search result](https://github.com/search?l=Dart&q=TextSelectionOverlay+fadeDuration&type=Code)

Would it be possible to add the following deprecated const value?

```dart
class TextSelectionOverlay {

  /// Controls the fade-in and fade-out animations for the toolbar and handles.
  @Deprecated(
      'Use `SelectionOverlay.fadeDuration` instead. '
      'This value was deprecated in Flutter 2.12.0-4.1.pre'
  )
  static const Duration fadeDuration = SelectionOverlay.fadeDuration;
}
```

**Expected results:** 

No build errors.

**Actual results:** 

An error occurs during build.

```
../../../.pub-cache/hosted/pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40:
Error: Member not found: 'fadeDuration'.
        duration: TextSelectionOverlay.fadeDuration, vsync: this);
```

<details>
<summary>Code sample</summary>

```dart
```

</details>

<details>
  <summary>Logs</summary>

```
../../../.pub-cache/hosted/[pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40](http://pub.dartlang.org/flutter_math_fork-0.6.1/lib/src/widgets/selection/handle_overlay.dart:60:40):
Error: Member not found: 'fadeDuration'.
        duration: TextSelectionOverlay.fadeDuration, vsync: this);
```

```
```

```
$ flutter doctor -v
[✓] Flutter (Channel beta, 2.12.0-4.1.pre, on macOS 12.3 21E230 darwin-arm,
    locale ja-JP)
    • Flutter version 2.12.0-4.1.pre at /Users/koji/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 680962aa75 (30 hours ago), 2022-03-16 13:46:03 -0700
    • Engine revision e9f57b5d0f
    • Dart version 2.17.0 (build 2.17.0-182.1.beta)
    • DevTools version 2.11.1

[!] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/koji/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 65.2.4
    • Dart plugin version 213.7227

[✓] VS Code (version 1.65.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 99.0.4844.74

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/100343

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text_selection.dart [hunk 1]
```diff
@@ -303,6 +303,13 @@ class TextSelectionOverlay {
     );
   }
 
+  /// Controls the fade-in and fade-out animations for the toolbar and handles.
+  @Deprecated(
+    'Use `SelectionOverlay.fadeDuration` instead. '
+    'This feature was deprecated after 2.12.0-4.1.pre.'
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/text_selection.dart:309]: You have an analyzer failure about this deprecated format.  Maybe it's just a missing trailing comma here?

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Was this removed before without being deprecated?

File: packages/flutter/lib/src/widgets/text_selection.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/lib/src/widgets/text_selection.dart:309]: yes

**Comment by Hixie** [comment]
Writing a test for this seems simple: just check that the property exists and has the value you expect. :-)

**Comment by chunhtai** [comment]
@Hixie I thought the analyzer will complain about using deprecated field in test? Also is it worth it to add a test for a to-be-deprecated field

**Review by Piinks — APPROVED** [review]
LGTM! Thanks for writing dart fixes!

**Pull request merged**